--- a/Makefile	2020-07-08 06:04:39.000000000 +0200
+++ a/Makefile	2026-01-17 16:58:27.172219992 +0100
@@ -41,6 +41,13 @@
 # Get SDL2 related build flags.
 SDL2_CFLAGS := $(shell sdl2-config --cflags) -DUSE_SDL2
 SDL2_LFLAGS := $(shell sdl2-config --static-libs || sdl2-config --libs)
+# above bails out with permission denied of multiarch
+# https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1051946
+# seems to be fixed in trixie 2.32.4+dfsg-1
+ifeq ($(shell echo "$(SDL2_LFLAGS)" | grep -c "//"),1)
+ _MULTIARCH := $(shell gcc -print-multiarch)
+ SDL2_LFLAGS := $(shell echo "$(SDL2_LFLAGS)" | sed -e s'#//#/$(_MULTIARCH)/#')
+endif
 
 # Override these in a .mak file. See buidlcfg/00-compiler*.txt for examples.
 #
